tracer/src/Datadog.Trace/DatabaseMonitoring/DatabaseMonitoringPropagator.cs

```diff
@@ -113,21 +113,33 @@ internal static string PropagateDataViaComment(DbmPropagationLevel propagationSt
         /// Currently only working for MSSQL (uses an instruction that is specific to it)
         /// </summary>
         /// <returns>True if the traceparent information was set</returns>
-        internal static bool PropagateDataViaContext(DbmPropagationLevel propagationLevel, IntegrationId integrationId, IDbConnection? connection, Span span)
+        internal static bool PropagateDataViaContext(DbmPropagationLevel propagationLevel, IntegrationId integrationId, IDbCommand command, Span span)
         {
-            if (propagationLevel != DbmPropagationLevel.Full || integrationId != IntegrationId.SqlClient || connection == null)
+            if (propagationLevel != DbmPropagationLevel.Full || integrationId != IntegrationId.SqlClient || command.Connection == null)
             {
                 return false;
             }
 
+            if (command.Connection.State != ConnectionState.Open)
+            {
+                if (Log.IsEnabled(LogEventLevel.Debug))
```

Comment:
why this check ? We are not doing any expensive operation on debug.

Reply:
Ah thanks! Just copy/pasting, removed it